src/realm/util/encrypted_file_mapping.cpp

```diff
@@ -109,20 +118,24 @@ off_t iv_table_pos(off_t pos)
     const size_t index = static_cast<size_t>(pos) / block_size;
     const size_t metadata_block = index / blocks_per_metadata_block;
     const size_t metadata_index = index & (blocks_per_metadata_block - 1);
-    return metadata_block * (blocks_per_metadata_block + 1) * block_size + metadata_index * metadata_size;
+    return off_t(metadata_block * (blocks_per_metadata_block + 1) * block_size + metadata_index * metadata_size);
 }
 
-void check_write(int fd, off_t pos, const void* data, size_t len)
+void check_write(FileDesc fd, off_t pos, const void* data, size_t len)
 {
-    ssize_t ret = pwrite(fd, data, len, pos);
-    REALM_ASSERT(ret >= 0 && static_cast<size_t>(ret) == len);
+	uint64_t orig = File::get_file_pos(fd);
```

Comment:
I'll look in to that later. Because it's unclear from MS documentation if it will preserve the original file position, and I'm also not sure if there are places in Core that rely on it. Made an issue here https://github.com/realm/realm-core/issues/2615